include/ginkgo/core/matrix/dense.hpp

```diff
@@ -827,14 +827,14 @@ class Dense
     }
 
     /**
-     * Creates a Dense matrix with the same configuration as the callers matrix.
+     * Creates a Dense matrix with the same size as the callers matrix.
+     * Note that the stride is not preserved!
```

Comment:
We basically have a choice here between
1. adding a new function create_with_same_size, keeping the original behavior in create_with_same_config
2. modifying create_with_same_config to discard original stride
3. keep behavior as-is

1. breaks MFEM support, 2. changes behavior slightly, 3. can give bad memory usage and performance when applying solvers on a system with huge padding, and makes the `compact` optimization pretty much impossible, so I'd probably need to add something specific like `run_solver_kernel` expicitly that provides the correct stride in the device_unpack step.

Reply:
I think we should still go for 1. We have not yet released Ginkgo 1.4.0. So we should fix this now so that we dont have something whose name says that we preserve the stride. We can create a small PR on MFEM's side as well.